Deflect server optimizations

hernando opened the pull request:
According to VTune, QMetaObject::activate from the emit receivedTile inside
ServerWorker::_handleMessage was hotspot #1. Using a direct connection
removes it.
The other optimization is trivial, but it was another significant hotspot.

Commit:
Deflect server optimizations

According to VTune, QMetaObject::activate from the emit receivedTile inside
ServerWorker::_handleMessage was hotspot #1. Using a direct connection
removes it. Thread safety had to be revisited under this new scenario.

The other optimization is trivial, but it was another significant hotspot.

hernando commented:
I haven't checked yet whether the direct connection is thread safe.

hernando commented:
Now I did. It looks safe to me, but we can discuss it once you're back.

hernando commented:
[A test has failed](https://travis-ci.org/BlueBrain/Deflect/jobs/501544515) and it could be a thread related thing because I've tried to reproduce it and I've got random errors in ServerTests, but not that one in particular. I even tried makin the ServerWorker::receivedFrameFinished to FrameDispatcher::processFrameFinished connection direct, but it didn't help.

dnachbaur commented on deflect/server/Server.cpp:
```diff
@@ -125,7 +125,7 @@ class Server::Impl : public QTcpServer
                     &ServerWorker::closeConnection);
             connect(worker, &ServerWorker::receivedTile, frameDispatcher,
-                    &FrameDispatcher::processTile);
+                    &FrameDispatcher::processTile, Qt::DirectConnection);
```
can't judge about concurrent access that happens now that the tiles are added one a different thread. they're probably save, as the finish of a frame is still queued?

Commit:
Corrections in ServerTests

- Fixture 2 is not guaranteed to get all frames
- Race condition between requestFrame and finish frame message handler.
  Requests forwarded through DeflectServer thread.

Commit:
Use direct connection to forward disconnect signal.

To make the timing of the signal more predictable during unexpected
disconnects.

hernando commented:
I would merge this as is for the moment, it's one step further, even if not all issues are resolved.
